main.py

```diff
@@ -368,6 +385,26 @@ def test_new_finalised_node_pruning():
     assert tree.size == 4
 
 
+def test_ghost():
+    # Setup
+    genesis = Block(None)
+    tree = CompressedTree(genesis)
+
+    for i in range(3):
+        block = Block(genesis, 1)
+        _ = tree.add_new_latest_block(block, i)
+
+    val_0_block = tree.latest_block_nodes[0].block
+    for i in range(3):
+        block = Block(val_0_block, 1)
+        _ = tree.add_new_latest_block(block, i)
+
+    val_0_block = tree.latest_block_nodes[0].block
+    # Giving this block more weight, gives GHOST determanism
+    head_node = tree.add_new_latest_block(Block(val_0_block, weight=2), i)
```

Comment:
by `i`, you mean `2` right?

Reply:
Yes, that `i` shouldn't be there, well caught. This is the problem with copy pasta.

```suggestion
    head_node = tree.add_new_latest_block(Block(val_0_block, weight=2), 0)
```